pyomo/solvers/plugins/solvers/cplex_direct.py

```diff
@@ -356,7 +357,11 @@ def _add_sos_constraint(self, con):
 
         self._vars_referenced_by_con[con] = ComponentSet()
 
-        for v, w in con.get_items():
+        if hasattr(con, 'get_items'):
+            sos_items = con.get_items()
+        else:
+            sos_items = con.items()
```

Comment:
What is `get_items`?  Is there a reason we don't just use `.items()`, or better yet, `six.iteritems(con)`?

Reply:
get_items() applies when it is an SOSConstraint coming from the ASL. items() applies when it is an sos object from the kernel (where I've attempted to clean up the interface for various components, which sometimes requires little hacks like these in the writers until (hopefully) the AML adopts it)

`six.iteritems(con)` (for kernel sos): It is not implemented as a fully dict-like thing, though it certainly could be. I just went ahead and kept the implementation as simple as possible. So this would not work.